api/lifecycle.go

```diff
@@ -0,0 +1,32 @@
+// NOTE: Subject to change, do not rely on this package from outside git-lfs source
+package api
+
+import "net/http"
+
+// TODO: extract interface for *http.Request; update methods. This will be in a
+// later iteration of the API client.
+
+// A Lifecycle represents and encapsulates the behavior on an API request from
+// inception to cleanup.
+//
+// At a high level, it turns an *api.RequestSchema into an
+// api.Response (and optionally an error). Lifecycle does so by providing
+// several more fine-grained methods that are used by the client to manage the
+// lifecycle of a request in a platform-agnostic fashion.
+type Lifecycle interface {
+	// Build creates a sendable request by using the given RequestSchema as
+	// a model.
+	Build(req *RequestSchema) (*http.Request, error)
+
+	// Execute transforms generated request into a wrapped repsonse, (and
+	// optionally an error, if the request failed), and serializes the
+	// response into the `into interface{}`, if one was provided.
+	Execute(req *http.Request, into interface{}) (Response, error)
```

Comment:
I thought part of the idea of abstracting this was to make the interface non-specific to HTTP, but you're including `http.Request` in the interface here. `Response` conversely is non-protocol specific, which is more what I expected.